apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLoggerExtension.cs

```diff
@@ -98,33 +78,21 @@ public static void UseGoogleExceptionLogging(this IApplicationBuilder app)
         /// </para>
         /// </summary>
         /// <param name="services">The service collection. Cannot be null.</param>
-        /// <param name="serviceName">An identifier of the service, such as the name of the 
-        ///     executable or job. Cannot be null.</param>
-        /// <param name="version">Represents the source code version that the developer 
-        ///     provided. Cannot be null.</param> 
-        /// <param name="options">Error reporting options for exception logging.</param>   
+        /// <param name="setupAction">Action to set up options. Cannot be null.</param>
         public static void AddGoogleExceptionLogging(
-            this IServiceCollection services, string serviceName, string version,
-            ErrorReportingOptions options = null) =>
-                AddGoogleExceptionLoggingBase(services, null, serviceName, version, options);
+            this IServiceCollection services, Action<ErrorReportingServiceOptions> setupAction)
+        {
+            GaxPreconditions.CheckNotNull(services, nameof(services));
+            GaxPreconditions.CheckNotNull(setupAction, nameof(setupAction));
 
+            var serviceOptions = new ErrorReportingServiceOptions();
+            setupAction(serviceOptions);
+            var serviceName = GaxPreconditions.CheckNotNull(serviceOptions.ServiceName, nameof(serviceOptions.ServiceName));
+            var version = GaxPreconditions.CheckNotNull(serviceOptions.Version, nameof(serviceOptions.Version));
 
-        /// <summary>
-        /// Shared code for creating error reporting services.
-        /// </summary>
-        /// <param name="services">The service collection. Cannot be null.</param>
-        /// <param name="projectId">The Google Cloud Platform project ID. If null the project Id will be auto detected.</param>
-        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.
-        ///     Cannot be null.</param>
-        /// <param name="version">Represents the source code version that the developer provided. 
-        ///     Cannot be null.</param>
-        /// <param name="options">Optional, error reporting options.</param>
-        private static void AddGoogleExceptionLoggingBase(
-            this IServiceCollection services, string projectId, string serviceName, string version,
-            ErrorReportingOptions options = null)
-        {
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
-            services.AddSingleton(ErrorReportingContextExceptionLogger.Create(projectId, serviceName, version, options));
+            services.AddSingleton(ErrorReportingContextExceptionLogger.Create(
+                serviceOptions.ProjectId, serviceName, version, serviceOptions.Options));
```

Comment:
No, it's called further down in the stack as it's not required that a PID is used.   A user could log to an organization (this is not common as far as I've seen).

The PID option is a convince as this is how most users will probably log.